httpcore/_async/connection.py

```diff
@@ -81,103 +68,25 @@ async def arequest(
         ext: dict = None,
     ) -> Tuple[int, Headers, AsyncByteStream, dict]:
         assert url_to_origin(url) == self.origin
-        ext = {} if ext is None else ext
-        timeout = cast(TimeoutDict, ext.get("timeout", {}))
-
-        async with self.request_lock:
-            if self.state == ConnectionState.PENDING:
-                if not self.socket:
-                    logger.trace(
-                        "open_socket origin=%r timeout=%r", self.origin, timeout
-                    )
-                    self.socket = await self._open_socket(timeout)
-                self._create_connection(self.socket)
-            elif self.state in (ConnectionState.READY, ConnectionState.IDLE):
-                pass
-            elif self.state == ConnectionState.ACTIVE and self.is_http2:
-                pass
-            else:
-                raise NewConnectionRequired()
```

Comment:
I'm not sure about why this `NewConnectionRequired()` case was introduced, so it's possible the new impl is missing something in the HTTP/1.1 case.

Reply:
I was also unsure what this was for, a quick glance at the repo and I couldn't find any location where it was caught to be handled.